Tasks/Common/kubernetes-common-v2/image-metadata-helper.ts

```diff
@@ -0,0 +1,246 @@
+                    // Filter all pods using the container names in this deployment,
+                    // and get the imageIds from pod status
+                    imageIds = getImageIdsForPodsInDeployment(container.name, allPods.items);
+                });
+            }
+        }
+    }
+    catch (e) {
+        // Don't fail the task if the image ID extraction fails
+        console.log("Image Ids extraction failed with exception: " + e);
+    }
+
+    let name: string = deploymentObject.metadata && deploymentObject.metadata.name ? deploymentObject.metadata.name : "";
+    let relatedUrls = [getPipelineUrl()];
+    let clusterUrl = getServerUrl(clusterInfo);
+    if (clusterUrl) {
+        relatedUrls.push(clusterUrl);
+    }
+
+    if (manifestFilePaths) {
+        relatedUrls.push(...manifestFilePaths);
+    }
+    else {
+        let manifestPaths = getManifestFilePaths();
+        if (manifestPaths.length > 0) {
+            relatedUrls.push(...manifestPaths);
+        }
+    }
+
+    const metadataDetails = {
+        "Name": name,
+        "Description": getDescription(),
+        "RelatedUrl": relatedUrls,
+        "ResourceUri": imageIds,
+        "UserEmail": getUserEmail(),
+        "Config": deploymentStrategy,
+        "Address": getEnvironmentResourceAddress() || clusterUrl,
+        "Platform": getPlatform()
+    };
+
+    return metadataDetails;
+}
+
+export function getImageIdsForPodsInDeployment(containerName: string, pods: any[]): string[] {
+    // The image name in parent.spec.template.spec.containers and in pod.status.containerStatuses is not a constant, example it is redis in former, and redis:latest in latter
+    // Hence filtering the pods on the basis of container name which is a constant
+    let imageIds: string[] = [];
+    pods.forEach(pod => {
+        const podStatus = pod.status;
+        podStatus.containerStatuses.forEach(status => {
+            if (status.name.toLowerCase() === containerName.toLowerCase()) {
+                if (status.imageID) {
+                    imageIds.push(getImageResourceUrl(status.imageID));
+                }
+            }
+        });
+    });
+
+    return imageIds;
+}
+
+export function getImageIdsForPod(pod: any): string[] {
+    let imageIds: string[] = [];
+    const podStatus = pod.status;
+    podStatus.containerStatuses.forEach(status => {
+        if (status.imageID) {
+            imageIds.push(getImageResourceUrl(status.imageID));
+        }
+    });
+
+    return imageIds;
+}
+
+export function getImageResourceUrl(imageId: string): string {
+    const sha256Text = "@sha256:";
+    const separator = "://";
+    let indexOfSeparator = imageId.indexOf(separator);
+    let image = indexOfSeparator >= 0 ? imageId.substr(indexOfSeparator + separator.length) : imageId;
+    const digest = getImageResourceUrlParameter(imageId, matchPatternForDigest);
+
+    let match = image.match(/^(?:([^\/]+)\/)?(?:([^\/]+)\/)?([^@:\/]+)(?:[@:](.+))?$/);
```

Comment:
Can we use dockercommonutils2 getResoruceName()?

Reply:
No, Two utils files cannot have dependency on each other. It throws build error